fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/loanschedule/domain/AbstractCumulativeLoanScheduleGenerator.java

```diff
@@ -753,6 +778,24 @@ private void updateCompoundingDetails(final Collection<LoanScheduleModelPeriod>
         }
     }
 
+    private void applyChargesForCurrentPeriodOnScheduleGenerate(final Set<LoanCharge> loanCharges, final MonetaryCurrency currency,
```

Comment:
@tabrez-fiter Why are we creating a new method when it is doing exactly the same thing asapplyChargesForCurrentPeriod methdod?

Reply:
The previous method modified julius made the generator broken so I replicate this to made it work

Reply:
But both methods are doing the same thing. How will it avoid breaking the generator?

Reply:
Also please remove the log statements

Reply:
@faheem205  
1. This method is not copy  of applyChargesForCurrentPeriod , the fixes I did was working as per expectation  until recent merge for SU-527 , if you compare these 2 method you can find the difference.
2. Can you be specific if any particular log you are having concern , I can change the log to debug though. its for future troubleshooting.

Reply:
it was breaking on some existing loans (from UAT ) if you run the job on code from SU-527 
This is just to avoid any break due to last minute change 
![image](https://github.com/user-attachments/assets/702db61c-719e-4dc7-be5c-a8c179c9d62c)